File: tournaments/stringsConstruction/stringsConstruction.cs
int stringsConstruction(string a, string b) {
    var result = int.MaxValue;
    var aStore = new int[26];
    var bStore = new int[26];
    
    for (int i = 0; a.Length > i; ++i) {
        ++aStore[a[i] - 'a'];
    }
    
    for (int i = 0; b.Length > i; ++i) {
        ++bStore[b[i] - 'a'];
    }
    
    for (int i = 0; a.Length > i; ++i) {
        if (aStore[a[i] - 'a'] > bStore[a[i] - 'a']) {
            return 0;
        } else {
            result = Math.Min(result, bStore[a[i] - 'a'] / aStore[a[i] - 'a']);
        }
    }
    
    return result;
}

File: tournaments/stringsConstruction/stringsConstruction.js
function stringsConstruction(a, b) {
    const aStore = Array(26).fill(0);
    const bStore = Array(26).fill(0);
    let result = Number.MAX_SAFE_INTEGER;

    for (const letter of a) {
        ++aStore[letter.charCodeAt() % 26];
    }

    for (const letter of b) {
        ++bStore[letter.charCodeAt() % 26];
    }

    for (const letter of a) {
        const aCount = aStore[letter.charCodeAt() % 26];
        const bCount = bStore[letter.charCodeAt() % 26];

        if (aCount > bCount) {
            return 0;
        } else {
            result = Math.min(result, ~~(bCount / aCount));
        }
    }

    return result;
}
